internal/descriptor/cache/key_test.go

```diff
@@ -0,0 +1,93 @@
+package cache_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/descriptor/cache"
+)
+
+func TestGenerateDescriptorCacheKey(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		template *v1beta2.ModuleTemplate
+		want     cache.DescriptorCacheKey
+	}{
+		{
+			name: "Annotations is not nil and valid semver",
+			template: &v1beta2.ModuleTemplate{
```

Comment:
I suggest creating a local ModuleTemplate generator to abstract-out the internal structure of the module template that is irrelevant to the unit test:

```go
func buildModuleTemplate(name, annotation, channel string) *v1beta2.ModuleTemplate {
   // ...
}
```

Reply:
Using the utils builder 👍